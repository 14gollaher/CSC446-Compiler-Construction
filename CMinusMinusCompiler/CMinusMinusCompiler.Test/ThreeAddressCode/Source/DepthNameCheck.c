int a;
int foo()
{
	int a;
	return a * a + a;
}

int b;
int main()
{
	int a;

	a = a + a * a;
	a = foo(a, b);
	return a;
}