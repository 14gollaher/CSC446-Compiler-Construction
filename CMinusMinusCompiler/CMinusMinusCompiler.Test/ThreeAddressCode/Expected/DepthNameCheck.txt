PROC foo
	_BP-4 = _BP-2 * _BP-2
	_BP-6 = _BP-4 + _BP-2
	_AX = _BP-6
ENDP foo

PROC main
	_BP-6 = _BP-2 * _BP-2
	_BP-4 = _BP-2 + _BP-6
	_BP-2 = _BP-4
	_PUSH _b
	_PUSH _BP-2
	_CALL foo
	_BP-2 = _AX
	_AX = _BP-2
ENDP main

START PROC main